### PR TITLE
Restructuring OnResetTimeout

### PR DESCRIPTION
**Pull request by atiwari9**
This proposal is to use OnResetTimeout for applicable interfaces so as to update the function and expand it to be used by a broader set of RPCs than just in _UI_ and _TTS_ interfaces.

### CODE REVIEW

**Comment by atiwari9** [comment]
@theresalech  - This proposal is ready to be reviewed

File: proposals/NNNN-Restructuring-OnResetTimeout.md [hunk 1]
```diff
@@ -0,0 +1,122 @@
+# Restructuring OnResetTimeout
+
+* Proposal: [SDL-NNNN](NNNN-Restructuring-OnResetTimeout.md)
+* Author: [Ankur Tiwari](https://github.com/ATIWARI9)
+* Status: **Awaiting review**
+* Impacted Platforms: [Core / RPC]
+
+## Introduction
+
+This proposal is to use OnResetTimeout for applicable interfaces so as to update the function and expand it to be used by broader set of RPCs than just in _UI_ and _TTS_ interfaces.
```
Comment (theresalech) [inline proposals/NNNN-Restructuring-OnResetTimeout.md]: I think this should be updated to "...by a broader set of RPCs..." This should be updated in the PR description as well.

File: proposals/NNNN-Restructuring-OnResetTimeout.md [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+    <param name="extensionPeriod" type="Integer" minvalue="0" maxvalue="1000000" mandatory="false">
+		<description>
+			Timeout period in milliseconds, for the method for which timeout needs to be reset.
+			If omitted, timeout would be reset by defaultTimeout specified in smartDeviceLink.ini
+		</description>
+    </param>
+</function>
+```
+
+**Note**: Though type needs to be Long for _extensionPeriod_, i have kept it as Integer to maintain consistency in API as no other param uses Long irrespective of _maxvalue_
```
Comment (theresalech) [inline proposals/NNNN-Restructuring-OnResetTimeout.md]: Please capitalize "I"

**Comment by atiwari9** [comment]
@theresalech  - Updated the PR as per your comments.

**Pull request merged**